src/java.base/unix/classes/sun/nio/fs/UnixCopyFile.java

```diff
@@ -248,17 +299,51 @@ private static void copyFile(UnixPath source,
             // set to true when file and attributes copied
             boolean complete = false;
             try {
-                // transfer bytes to target file
-                try {
-                    long comp = Blocker.begin();
+                boolean transferred = false;
+                if (!transferNotSupported) {
+                    // transfer bytes to target file
                     try {
-                        transfer(fo, fi, addressToPollForCancel);
-                    } finally {
-                        Blocker.end(comp);
+                        int res;
+                        do {
+                            long comp = Blocker.begin();
+                            try {
+                                res = transfer0(fo, fi, addressToPollForCancel);
+                            } finally {
+                                Blocker.end(comp);
+                            }
+                        } while (res == IOStatus.INTERRUPTED);
```

Comment:
As `RESTARTABLE` is used in `transfer0()` this loop could go away.

Reply:
> As `RESTARTABLE` is used in `transfer0()` this loop could go away.

Does the check for EINTR in transfer0 also go away? It maps it to IOS_INTERRUPTED but the use of RESTARTABLE means this code will never execute.

Reply:
Yes, as noted [here](https://github.com/openjdk/jdk/pull/9161#discussion_r900290005).